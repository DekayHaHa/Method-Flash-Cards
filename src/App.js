import React, { Component } from 'react';
import methodCards from './method-cards'
import Header from './Header';
import Container from './Container';

export default class App extends Component {
  constructor() {
    super ();
    this.state = {
      correct: [],
      questions: []
    }
  }
  keepScore = (question) => {
    const newScore = this.state.correct;
    newScore.push(question);
    this.setState({ correct: newScore}, () => {
      localStorage.setItem('correct', JSON.stringify(newScore));
    })
  }
  componentWillMount() {
    let score = JSON.parse(localStorage.getItem('correct')) || [];
    if(score.length){
      this.setState({correct: score, questions: this.filterStorage(score)})
    } else {
      this.setState({questions: methodCards})
    }
  }
  filterStorage = (score) => {
    return methodCards.filter(val => {
      return score.every(cQ => cQ.uid !== val.uid)
    })
  }
  render() {
    return (
      <section>
        <Header/>
        <Container cards={this.state.questions}
          keepScore={this.keepScore}/>
      </section>
    );
  }
}

