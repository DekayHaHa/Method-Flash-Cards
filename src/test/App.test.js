import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';

const mockData = [{
  "snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "((acc, val) => acc += val)", "console.log(results) // logs 15"],
  "category": "Array Prototype Method",
  "solution": "reduce",
  "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce",
  "uid": "card-10"
},
{
  "snippet": ["let str = 'Burritos are the best!'", "let results = str.", "('best!')", "console.log(results) // logs true"],
  "category": "String Prototype Method",
  "solution": "endsWith",
  "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith",
  "uid": "card-11"
},
{
  "snippet": ["let str = 'Burritos are the best times 10!'", "let results = str.", "(10)", "console.log(results) // logs [ '10', index: 28, input: 'Burritos are the best times 10!' ]"],
  "category": "String Prototype Method",
  "solution": "match",
  "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match",
  "uid": "card-12"
}]
const mockScore = [{
  "snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "((acc, val) => acc += val)", "console.log(results) // logs 15"],
  "category": "Array Prototype Method",
  "solution": "reduce",
  "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce",
  "uid": "card-10"
},
{
  "snippet": ["let str = 'Burritos are the best times 10!'", "let results = str.", "(10)", "console.log(results) // logs [ '10', index: 28, input: 'Burritos are the best times 10!' ]"],
  "category": "String Prototype Method",
  "solution": "match",
  "mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match",
  "uid": "card-12"
}]

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App/>
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      questions: [],
      roundReset: false
    });
  });
  it('Should set reset state to true', () => {
    expect(wrapper.state('roundReset')).toEqual(false)
    wrapper.instance().reset()
    expect(wrapper.state('roundReset')).toEqual(true)
  })
  it('Should set reset state to true', () => {
    wrapper.instance().reset()
    expect(wrapper.state('roundReset')).toEqual(true)
    wrapper.instance().setToggle()
    expect(wrapper.state('roundReset')).toEqual(false)
  })
  it('should use original questions if nothing in local storage', () => {
    wrapper.setState({questons: mockData})
    expect(wrapper.instance().checkStorage(mockData)).toEqual(mockData)
    expect(wrapper).toMatchSnapshot();    
  })
  it('should filter out questions by uid', () => {
    expect(wrapper.instance().filterStorage(mockScore, mockData)).toEqual([mockData[1]])
  })
})
