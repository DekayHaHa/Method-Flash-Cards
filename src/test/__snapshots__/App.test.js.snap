// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshop 1`] = `
<section>
  <Container
    cards={Array []}
    reset={[Function]}
    resetCheck={false}
    resetToggle={[Function]}
  />
</section>
`;

exports[`App should use original questions if nothing in local storage 1`] = `
<section>
  <Container
    cards={Array []}
    reset={[Function]}
    resetCheck={false}
    resetToggle={[Function]}
  />
</section>
`;
